doc/manual/contract.md

```diff
@@ -0,0 +1,505 @@
+
+Example:
+~~~
+class MyClass
+	fun foo(x: Int) is abstract, ensure(x == 0)
+end
+
+class MySub
+	super MyClass
+
+	redef fun foo(x: Int) is no_contract
+end
+~~~
+
+| Class   | Condition of `foo(x)` |
+|---------|-----------------------|
+| MyClass | `i == 0`              |
+| MySub   | `none`                |
+
+## Execution
+
+When performing the evaluation of a contract, all routines and elements will be executed without the evaluation of contracts. The objective is to remove the potential risk of falling into an infinite loop of verification.
+
+Now we add a small test program to see the contract of `Stack` class in action:
+
+~~~
+var stack = new Stack[String]
+stack.push("Banana")
+stack.push("Coconut")
+# stack.get_at(5) # Causes an assertion failed because the given index is > size
+stack.pop
+stack.pop
+# stack.pop # Causes an assertion failed because the stack is empty
+# stack.size = -1 # Causes an assertion failed because a stack cannot have a size < 0
+~~~
+
+### Execution order
+
+~~~
+class A
+	invariant(i > 10)
+
+	var i: Int
+
+	init default(i: Int)
+	is
+		expect(i > 0)
+		ensure(self.i == i + 10)
+	do
+		_i = i + 10
+	end
+
+	fun foo(x: Int): Bool
+	is
+		expect( x > 0 )
+		ensure( result == true )
+	do
+		return true
+	end
+end
+
+var a = new A(2)
+a.foo(10)
+~~~
+
+This is the evaluation order of the `var a = new A(2)`:
+
+1- preconditions (`expect(i > 0)`)
+2- called property (execution of the method body)
+3- postconditions (`ensure(self.i = i + 10)`)
+4- invariant (`invariant(i > 10)`)
+
+This is the evaluation order of the `a.foo`:
+
+1- invariant (`invariant(i > 10)`)
+2- preconditions (`expect( x > 0 )`)
+3- called property (execution of the method body)
+4- postconditions (`result == true`)
+5- invariant (`invariant(i > 10)`)
+
+In redefinition when you make a call to `super` the contracts is not checked again.
+
+Example:
+~~~
+class A
+	fun print_expect: Bool
+	do
+		print "In expect A"
+		return true
+	end
+
+	fun foo
+	is
+		expect(print_expect)
+	do
+		print "In foo of A"
+	end
+end
+
+class B
+	super A
+
+	fun print_new_expect: Bool
+	do
+		print "In expect B"
+		return true
+	end
+
+	redef fun foo
+	is
+		expect(print_new_expect)
+	do
+		super
+		print "In foo of B"
+	end
+end
+
+var b = new B
+b.foo
+~~~
+
+This example gives the following output:
+~~~
+In expect B
+In foo of A
+In foo of B
+~~~
+
+Note this example also shows the fact that during elevations of `expect` in an inheritance frame when one of the `expect` conditions is satisfied we stop the verification.
+
+To summarize the evaluation order for public methods/attribute, is as follows:
+
+1- invariant
+2- preconditions (`expect`)
+3- called property
+4- postconditions (`ensure`)
+5- invariant
+
+For constructors only the invariant verification in entry (first step) is ignored.
+
+### Options
```

Comment:
Options for `nitc`? `nit`? or both?

Reply:
Options are available on both. I added a sentence to specify it.